Use os.homedir when calculating tmpdir

shortenda opened the pull request:
See: [issue 225](https://github.com/TypeStrong/ts-node/issues/225)

Note that instead of basing off username as I'd said there this is based off the homedir from node os.

I also wasn't sure how best to go about testing this change, if there's any recommendations on that I would happily add tests to accompany.

blakeembrey commented on src/index.ts:
```diff
@@ -149,7 +149,15 @@ export function register (options: Options = {}): () => Register {
   const project = options.project || DEFAULTS.project
-  const cacheDirectory = options.cacheDirectory || DEFAULTS.cacheDirectory || join(tmpdir(), 'ts-node')
+
+  function getTmpDir(): string {
```
Can you move this function outside of the main run loop and add a space after the function name to the parenthesis to match the rest of the project? Also, I'll double check to make sure I didn't miss it, but why base it on the home directory instead of username or cwd?

shortenda replied:
The reason for not using the username was mostly because there is a method to get the homedir provided by node, but not one for username. So I thought this would be the easiest way to get good behavior across platforms. Alternatively to find the username we could look through env variables, or somehow use [this libuv code](http://docs.libuv.org/en/v1.x/misc.html#c.uv_os_get_passwd) (not sure how easy the ffi would be for this...); but neither of these are things I feel like I could do well. I could attempt this if it is desirable though.

Additionally, I think that homedir should be a reasonable option. I don't think that shared homedir's will be a common environment, and even if there are some setups like that this will not be a regression.

I opted not to use the cwd because I thought it would be desirable to share the cache between projects run by the same user. Additionally, two users could run code from the same directory and have conflicting caches. The only case I can think of where this would be superior would be if the user's homedir is empty string, in which case using the cwd would probably be better. Could fall-back to using the cwd if the users homedir is empty string? At the cost of a little more complexity this should provide better behavior.

I will update the code per your comments.

Thanks,
David

blakeembrey replied:
No worries, it seems reasonable. I just wanted to find out why, thanks! 👍  There is a `userInfo` method, but it appears to be new in v6 so won't be very useful to us here.

shortenda commented:
I've addressed your comments in the most recent commit.

Would you mind if I open a pr in tslint-config-standard to add the rule about space after function names?

blakeembrey commented:
It looks good, thanks 👍 

> Would you mind if I open a pr in tslint-config-standard to add the rule about space after function names?

Absolutely. The rules all come from the `eslint` collection (https://www.npmjs.com/package/tslint-eslint-rules) and I don't believe that particular one has been implemented yet. Might be an easier one to implement 😄
